File: tests/nodejs/cases/case_086_route_subscription.js
/**
 * Task case 086: route_subscription
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case086RouteSubscription(input) {
  // Task: route subscription records and prepare transport-ready payload.
  const taskName = 'route_subscription';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'subscription',
    stage: 'route',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_086', payload);
  return 'ok';
}

module.exports = {
  case086RouteSubscription,
};

File: tests/python/cases/case_086_route_subscription.py
"""
Task case 086: route subscription payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_086_route_subscription(input_data):
    # Task: route subscription records and prepare transport-ready payload.
    task_name = 'route_subscription'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'subscription',
        'stage': 'route',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_086'] = payload
    return 'ok'
